src/formatDate.js

```diff
@@ -50,7 +50,36 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+
+  const dateParts = date.split(fromSeparator);
+
+  const mapping = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    mapping[fromFormat[i]] = dateParts[i];
+  }
+
+  if (mapping['YY'] && !mapping['YYYY']) {
+    if (parseInt(mapping['YY']) < 30) {
+      mapping['YYYY'] = '20' + mapping['YY'];
+    } else {
+      mapping['YYYY'] = '19' + mapping['YY'];
+    }
+  }
+
+  if (mapping['YYYY'] && !mapping['YY']) {
+    mapping['YY'] = mapping['YYYY'].slice(-2);
+  }
```

Comment:
```suggestion
if (mapping['YY']) {
    mapping['YYYY'] = parseInt(mapping['YY']) < 30
      ? '20' + mapping['YY']
      : '19' + mapping['YY'];
  }

  if (mapping['YYYY']) {
    mapping['YY'] = mapping['YYYY'].slice(-2);
  }
```
suggest to use ternary instead nested if/else
also looks like condition in if can be simplier